apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -124,4 +128,60 @@ defmodule Snitch.Domain.Order.Transitions do
   end
 
   def persist_shipment(%Context{valid?: false} = context), do: context
+
+  defp process_package(package, shipping_method_id) do
+    shipping_method =
+      Enum.find(package.shipping_methods, fn %{id: id} ->
+        id == shipping_method_id
+      end)
+
+    package_total =
+      Enum.reduce(
+        [shipping_method.cost],
+        &Money.add!/2
+      )
+
+    Package.update(package, %{
+      cost: shipping_method.cost,
```

Comment:
This would give error on `shipping_method.cost` if `shipping_method` is nil. This can't be handled in controller. It would require `try rescue block` then.

Reply:
Yes, we need guard functions in context or wrapper in web controller.
TMB, we are moving ahead.